### ISSUE TITLE
Define data-driven car types and special traits

### ISSUE THREAD

**Issue by Miragebell** [Silverware-Games/war-tainted-falls-issues#9]
Updated after code/GDD audit on 2026-05-19.

The old enemy/baddie framing is now car-wash themed. Current code has `ObjectTypes.CAR`, but `GameConfig.xml` still contains older `<baddie>` data and `gameConfig.js` currently parses towers only.

This issue tracks car/unit design and data support:
- Define car types such as rusty car, muddy truck, luxury sedan, fleet vehicles, etc.
- Give cars data-driven dirt/health, speed, score value, and special traits.
- Decide which old baddie ideas still map to car-wash gameplay.
- Parse car stats from config if they remain data-driven.
- Add boss/special event cars only if they fit the GDD.

**Comment by michaelplzno** [comment]
Yes, this is planned: there are a lot of enemy types including enemies that contain other enemies, and lots of different power ups for bad guys.

Also adding some matchyverse bad guys is always an idea I'll go for.

**Comment by Miragebell** [comment]
> Also adding some matchyverse bad guys is always an idea I'll go for.

Not all of them may be "bad guys", though. Just imagine the star from matchygocthy quickly goes through the river and gives the player some zoot. Such wow, much bonus!

**Comment by michaelplzno** [comment]
YESSSSSSS

**Comment by michaelplzno** [comment]
2026-05-23 calendar/live-ops car note:

Special car types should be able to appear based on a real calendar, not only fixed wave progression. This could support a 365-day calendar of action: weekday-specific cars, holiday-themed cars, wacky observance days, famous birthdays/death days, movie release anniversaries, and other one-day or limited-window events.

Example direction:
- St. Paddy's Day could bring green cars, leprechaun-themed special units, luck/gold/Zoot modifiers, or themed bonus objectives.
- Certain weekdays could favor certain car families or grime traits.
- Famous anniversaries could temporarily unlock joke/event cars or themed wave pressure.

Design implication: car definitions should be data-driven enough that live-ops/calendar events can select, reskin, weight, buff, or introduce special cars without requiring every event to become a hardcoded one-off.

Related: #60 for timed environmental events/daily goals and #45 for rulesets/live-ops activation windows.
